codegen-server/src/test/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerEnumGeneratorTest.kt

```diff
@@ -0,0 +1,222 @@
+
+class ServerEnumGeneratorTest {
+    @Test
+    fun `it generates named enums`() {
+        val model = """
+            namespace test
+            @enum([
+                {
+                    value: "t2.nano",
+                    name: "T2_NANO",
+                    documentation: "T2 instances are Burstable Performance Instances.",
+                    tags: ["ebsOnly"]
+                },
+                {
+                    value: "t2.micro",
+                    name: "T2_MICRO",
+                    documentation: "T2 instances are Burstable Performance Instances.",
+                    tags: ["ebsOnly"]
+                },
+            ])
+            string InstanceType
+        """.asSmithyModel()
+        val provider = testSymbolProvider(model)
+        val writer = RustWriter.forModule("model")
+        val shape = model.lookup<StringShape>("test#InstanceType")
+        val generator = ServerEnumGenerator(model, provider, writer, shape, shape.expectTrait(), CodegenMode.Server, TestRuntimeConfig)
+        generator.render()
+        writer.compileAndTest(
+            """
+            let instance = InstanceType::T2Micro;
+            assert_eq!(instance.as_str(), "t2.micro");
+            assert_eq!(InstanceType::try_from("t2.nano").unwrap(), InstanceType::T2Nano);
+            // check no unknown
+            match instance {
+            InstanceType::T2Micro => (),
+            InstanceType::T2Nano => (),
+            }
+            """
+        )
+
+        writer.toString() shouldContain "#[non_exhaustive]"
+    }
+
+    @Test
+    fun `named enums are implement eq and hash`() {
+        val model = """
+            namespace test
+            @enum([
+            {
+                value: "Foo",
+                name: "Foo",
+            },
+            {
+                value: "Bar",
+                name: "Bar"
+            }])
+            string FooEnum
+        """.asSmithyModel()
+        val shape: StringShape = model.lookup("test#FooEnum")
+        val trait = shape.expectTrait<EnumTrait>()
+        val writer = RustWriter.forModule("model")
+        val generator = ServerEnumGenerator(model, testSymbolProvider(model), writer, shape, trait, CodegenMode.Server, TestRuntimeConfig)
+        generator.render()
+        writer.compileAndTest(
+            """
+            assert_eq!(FooEnum::Foo, FooEnum::Foo);
+            assert_ne!(FooEnum::Bar, FooEnum::Foo);
+            let mut hash_of_enums = std::collections::HashSet::new();
+            hash_of_enums.insert(FooEnum::Foo);
+            """.trimIndent()
+        )
+    }
+
+    @Test
+    fun `unnamed enums are implement eq and hash`() {
+        val model = """
+            namespace test
+            @enum([
+            {
+                value: "Foo",
+            },
+            {
+                value: "Bar",
+            }])
+            string FooEnum
+        """.asSmithyModel()
+        val shape: StringShape = model.lookup("test#FooEnum")
```

Comment:
```suggestion
        val shape = model.lookup<StringShape>("test#FooEnum")
```

Just to be consistent.